client/lib/credit-card-details/validation.js

```diff
@@ -68,54 +82,30 @@ validators.required = {
 	}
 };
 
-validators.validCreditCardNumber = creditCardValidator( 'validCardNumber' );
-
-validators.validCvvNumber = creditCardValidator( 'validCvc' );
-
-validators.validExpirationDate = creditCardValidator(
-	'notExpired',
-	'validExpirationMonth',
-	'validExpirationYear',
-);
-
-function validateCreditCard( cardDetails ) {
-	const expirationDate = cardDetails[ 'expiration-date' ] || '/',
-		expirationMonth = parseInt( expirationDate.split( '/' )[ 0 ], 10 ),
-		expirationYear = 2000 + parseInt( expirationDate.split( '/' )[ 1 ], 10 );
-
-	return creditcards.validate( {
-		number: cardDetails.number,
-		expirationMonth: expirationMonth,
-		expirationYear: expirationYear,
-		cvc: cardDetails.cvv
-	} );
-}
-
-function creditCardValidator( ...validationProperties ) {
-	return {
-		isValid: function( value, cardDetails ) {
-			if ( ! value ) {
```

Comment:
At the moment we use `required` check only when submitting the form. In effect we shouldn't validated all fields when user starts the process and fills in only one field and want to go to another one. We should only validate on focus change those field that are not empty.